src/main.rs

```diff
@@ -90,9 +88,21 @@ fn main() {
                 match command {
                     Ok(cmd) => {
                         debug!("Command: {:?}", cmd);
-                        let handler = commands::LogHandler { command: cmd.clone() };
+
+                        // Choose handler
+                        let handler: Box<Fn(&Command) -> Option<String> + Send> = match &*cmd.name {
```

Comment:
Here we explicitly need a boxed closure signature (essentially a trait object I think), otherwise we'd get an "incompatible return type" error between the arms of the expression.